WordPress/openverse
# Prevent accidentally committing focused test selectors

## Fixes



Follow up to https://github.com/WordPress/openverse/pull/2194 by @obulat 

## Description


The related PR removes an accidentally committed `.only` focused test selector. We can easily prevent accidental commits like that in the future with the Jest ESLint plugin and with Playwright's own `forbidOnly` setting.

This PR adds the Jest plugin and enables Playwright's `forbidOnly` setting in CI.

## Testing Instructions

Check out the changes made to appease the new ESLint plugin. I think they are good rules. I opened a discussion about whether relying on testing-library's `get*` function's error handling behaviour as an implicit assertion is a good idea long-term: https://github.com/WordPress/openverse/discussions/2198

For now I've just selectively disabled the `expect-expect` check or expanded the assert function list for modules that heavily rely on that.

The plugin caught a handful of tests that truly were incorrect, either because they had no assertions at all (!!) or because they were using `expect` incorrectly (like `expect(predicate)` rather than `expect(someValue).predicate()`, the former does nothing).

## Checklist


- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
